code/citadel/cit_arousal.dm

```diff
@@ -181,224 +181,365 @@
+	var/mb_time = 30
+	var/total_fluids = 0
+	var/datum/reagents/fluid_source = null
+
+	if(G.producing) //Can it produce its own fluids, such as breasts?
+		fluid_source = G.reagents
+	else
+		if(!G.linked_organ)
+			to_chat(src, "<span class='warning'>Your [G] is unable to produce it's own fluids, it's missing the organs for it.</span>")
+			return
+		fluid_source = G.linked_organ.reagents
+	total_fluids = fluid_source.total_volume
+
+	src.visible_message("<span class='danger'>[src] starts to [G.masturbation_verb] [p_their()] [G].</span>", \
+						"<span class='green'>You start to [G.masturbation_verb] your [G].</span>", \
+						"<span class='green'>You start to [G.masturbation_verb] your [G].</span>")
+
+	if(do_after(src, mb_time, target = src))
+		if(total_fluids > 5)
+			fluid_source.reaction(src.loc, TOUCH, 1, 0)
+			fluid_source.clear_reagents()
+		src.visible_message("<span class='danger'>[src] orgasms, cumming[istype(src.loc, /turf/open/floor) ? " onto [src.loc]" : ""]!</span>", \
+							"<span class='green'>You cum[istype(src.loc, /turf/open/floor) ? " onto [src.loc]" : ""].</span>", \
+							"<span class='green'>You have relieved yourself.</span>")
+		if(G.can_climax)
+			setArousalLoss(min_arousal)
+
+
+/mob/living/carbon/human/proc/mob_climax_outside(obj/item/organ/genital/G) //This is used for forced orgasms and other hands-free climaxes
+	var/mb_time = 30
+	var/total_fluids = 0
+	var/datum/reagents/fluid_source = null
+	var/unable_to_come = FALSE
+
+	if(G.producing) //Can it produce its own fluids, such as breasts?
+		fluid_source = G.reagents
+		total_fluids = fluid_source.total_volume
+	else
+		if(!G.linked_organ)
+			unable_to_come = TRUE
+		else
+			fluid_source = G.linked_organ.reagents
+			total_fluids = fluid_source.total_volume
+
+	if(unable_to_come)
+		src.visible_message("<span class='danger'>[src] shudders, their [G] unable to cum.</span>", \
+							"<span class='userdanger'>Your [G] cannot cum, giving no relief.</span>", \
+							"<span class='userdanger'>Your [G] cannot cum, giving no relief.</span>")
+	else
+		total_fluids = fluid_source.total_volume
+		src.visible_message("<span class='danger'>[src] looks like they're about to cum.</span>", \
+							"<span class='green'>You feel yourself about to orgasm.</span>", \
+							"<span class='green'>You feel yourself about to orgasm.</span>")
+		if(do_after(src, mb_time, target = src))
+			if(total_fluids > 5)
+				fluid_source.reaction(src.loc, TOUCH, 1, 0)
+			fluid_source.clear_reagents()
+			src.visible_message("<span class='danger'>[src] orgasms[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""], using [p_their()] [G]!</span>", \
+								"<span class='green'>You climax[istype(src.loc, /turf/open/floor) ? ", spilling onto [src.loc]" : ""] with your [G].</span>", \
+								"<span class='green'>You climax using your [G].</span>")
+		if(G.can_climax)
```

Comment:
You forgot a tab here. Without that tab, your arousal will reset even if the timer gets interrupted.